modules/queue/src/main/scala/ch/epfl/bluebrain/nexus/service/queue/KafkaSupervisedConsumer.scala

```diff
@@ -0,0 +1,102 @@
+package ch.epfl.bluebrain.nexus.service.queue
+
+import java.util.concurrent.TimeUnit.SECONDS
+
+import akka.actor.Actor.emptyBehavior
+import akka.actor._
+import akka.kafka.ConsumerMessage.{CommittableMessage, CommittableOffsetBatch, GroupTopicPartition}
+import akka.kafka.scaladsl.Consumer
+import akka.kafka.{ConsumerSettings, Subscriptions}
+import akka.persistence.query.Sequence
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import ch.epfl.bluebrain.nexus.service.indexer.persistence.IndexFailuresLog
+import ch.epfl.bluebrain.nexus.service.indexer.retryer.RetryOps._
+import ch.epfl.bluebrain.nexus.service.indexer.retryer.RetryStrategy.Backoff
+import io.circe._
+import io.circe.parser._
+import journal.Logger
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+/**
+  * Do not instantiate this class directly, see [[KafkaConsumer.start]].
+  */
+class KafkaSupervisedConsumer[Event](settings: ConsumerSettings[String, String],
+                                     index: Event => Future[Unit],
+                                     topic: String,
+                                     decoder: Decoder[Event])
+    extends Actor {
+
+  private implicit val as: ActorSystem       = context.system
+  private implicit val ec: ExecutionContext  = as.dispatcher
+  private implicit val mt: ActorMaterializer = ActorMaterializer()
+
+  private val log     = Logger[this.type]
+  private val config  = context.system.settings.config.getConfig("indexing.retry")
+  private val retries = config.getInt("max-count")
+  private val strategy =
+    Backoff(Duration(config.getDuration("max-duration", SECONDS), SECONDS), config.getDouble("random-factor"))
+  private val failures = IndexFailuresLog(s"kafka-consumer-$topic")
+
+  override val receive: Receive = emptyBehavior
+
+  override def preStart(): Unit = {
+    val done = Consumer
+      .committableSource(settings, Subscriptions.topics(topic))
```

Comment:
could we lift the option of using a committableSource or nonCommittableSource as a configuration? such that when we'd like to rebuild an "index" we can restart from the first event?